### PR TITLE
Merge IOHK fork

### PR DESCRIPTION
**Pull request by peterbecich**
Closes https://github.com/eskimor/purescript-bridge/issues/63

Uses https://hackage.haskell.org/package/wl-pprint-text-1.2.0.2

Merge is not complete yet

### CODE REVIEW

File: src/Language/PureScript/Bridge/Printer.hs [hunk 1]
```diff
@@ -114,434 +234,567 @@ _argonautCodecsImports settings
...
 
-importLineToText :: ImportLine -> Text
-importLineToText = \case
-    ImportLine importModule Nothing importTypes ->
-        "import " <> importModule <> " (" <> typeList importTypes <> ")"
-    ImportLine importModule (Just importAlias) _ ->
-        "import " <> importModule <> " as " <> importAlias
-  where
-    typeList s = T.intercalate ", " (Set.toList s)
+-- importLineToText :: ImportLine -> Text
```
Comment (peterbecich) [inline src/Language/PureScript/Bridge/Printer.hs:245]: TODO

File: src/Language/PureScript/Bridge/Printer.hs [hunk 2]
```diff
@@ -114,434 +234,567 @@ _argonautCodecsImports settings
...
+    lenses = "\n" <> sep <> "\n" <> T.pack (show (sumTypeToOptics st)) <> sep
     sep = T.replicate 80 "-"
 
-sumTypeToTypeDecls :: Switches.Settings -> SumType 'PureScript -> Text
-sumTypeToTypeDecls settings (SumType t cs is) =
-    T.unlines $
-        dataOrNewtype <> " " <> typeInfoToText True t <> " ="
-            : "    " <> T.intercalate "\n  | " (map (constructorToText 4) cs) <> "\n"
-            : instances settings (SumType t cs (filter genForeign . filter genArgonautCodec $ is))
+-- sumTypeToTypeDecls :: Switches.Settings -> SumType 'PureScript -> Text
```
Comment (peterbecich) [inline src/Language/PureScript/Bridge/Printer.hs:265]: TODO

File: .github/workflows/haskell.yml [hunk 3]
```diff
@@ -65,6 +65,6 @@ jobs:
     - name: Build
       run: |
           cabal build all --enable-tests --disable-optimization 2>&1 | tee build.log
-    - name: Test
-      run: |
-          cabal test all --disable-optimization
+    # - name: Test
```
Comment (peterbecich) [inline .github/workflows/haskell.yml:68]: TODO

**Comment by peterbecich** [comment]
TODO `DecodeJsonField`
https://github.com/eskimor/purescript-bridge/commit/c55622e325a422fbfc2101b0ed31d882fc55ceaf
https://github.com/eskimor/purescript-bridge/commit/c0dbf8626716552c924093e6d44a3b2426e5f4af

**Comment by peterbecich** [comment]
TODO fix type error in `example`. See `spago build`

**Comment by peterbecich** [comment]
TODO https://github.com/eskimor/purescript-bridge/issues/74#issuecomment-1498684546

**Comment by flip111** [comment]
When i load the code into GHCI i get

```
Not in scope: type constructor or class ‘Eq’
Not in scope: type constructor or class ‘Ord’
Not in scope: type constructor or class ‘Show’
```

for file https://github.com/peterbecich/purescript-bridge/blob/f1c0879094be41b3e6ef6846db00a28e461ef35a/src/Language/PureScript/Bridge/TypeInfo.hs

```haskell
import Data.Eq (Eq)
```

fixed it for me

**Comment by peterbecich** [comment]
@flip111 , I don't see that error.

Are you on GHC 9.2 or higher? If not, would you try using the Nix Flake?

`nix develop` will provide GHC 9.2. Then `cabal repl`.

**Comment by peterbecich** [comment]
More detail about https://github.com/eskimor/purescript-bridge/pull/77#issuecomment-1529154367

Using the parent repository currently, Haskell
```
data Bar a = Bar a
  deriving (Generic, Show, Typeable, FromJSON, ToJSON)
```

generates PureScript
```
newtype Bar a =
    Bar a

instance encodeJsonBar :: (Generic a ra, EncodeJson a) => EncodeJson (Bar a) where
  encodeJson = genericEncodeAeson Argonaut.defaultOptions
instance decodeJsonBar :: (Generic a ra, DecodeJson a, DecodeJsonField a) => DecodeJson (Bar a) where
  decodeJson = genericDecodeAeson Argonaut.defaultOptions
  
_Bar :: forall a. Iso' (Bar a) a
_Bar = _Newtype
```
That is correct.


In this PR,

```
data BarSimple a = BarSimple a
  deriving (Generic, Show, Typeable, FromJSON, ToJSON)
```

generates PureScript
```
newtype BarSimple a = BarSimple a

instance (Show a) => Show (BarSimple a) where
  show a = genericShow a

derive instance (Eq a) => Eq (BarSimple a)

derive instance (Ord a) => Ord (BarSimple a)

derive instance Newtype (BarSimple a) _

derive instance Generic (BarSimple a) _

instance (EncodeJson a) => EncodeJson (BarSimple a) where
  encodeJson = genericEncodeAeson Argonaut.defaultOptions

instance (DecodeJson a) => DecodeJson (BarSimple a) where
  decodeJson = genericDecodeAeson Argonaut.defaultOptions
```

. `DecodeJsonField` is missing

**Comment by peterbecich** [comment]
replaced by https://github.com/eskimor/purescript-bridge/pull/89

**Pull request closed without merging**